c/newuoa_c.f90

```diff
@@ -56,8 +57,12 @@ subroutine newuoa_c(cobj_ptr, data_ptr, n, x, f, nf, rhobeg, rhoend, ftarget, ma
 
 ! Read the inputs and convert them to the Fortran side types
 x_loc = real(x, kind(x_loc))
-rhobeg_loc = real(rhobeg, kind(rhobeg_loc))
-rhoend_loc = real(rhoend, kind(rhoend_loc))
+if (.not. is_nan(rhobeg)) then
```

Comment:
Similar to what was mentioned before, I do not think it is necessary to check whether rhobeg, or rhoend is NaN. Thank you for taking a look.